src/set_theory/cardinal.lean

```diff
@@ -938,6 +938,16 @@ lemma add_lt_omega_iff {a b : cardinal} : a + b < ω ↔ a < ω ∧ b < ω :=
 lemma omega_le_add_iff {a b : cardinal} : ω ≤ a + b ↔ ω ≤ a ∨ ω ≤ b :=
 by simp only [← not_lt, add_lt_omega_iff, not_and_distrib]
 
+lemma nsmul_lt_omega_iff {n : ℕ} {a : cardinal} : n • a < ω ↔ n = 0 ∨ a < ω :=
```

Comment:
I think it might be worth adding the trivial consequence
```lean
lemma nonzero_nsmul_lt_omega_iff {n : ℕ} {a : cardinal} (hn : n ≠ 0) : n • a < ω ↔ a < ω :=
-- something like `or.imp_left?`
```
since that seems to be what you would use in your main lemma.

Reply:
I ended up calling it `nsmul_lt_omega_iff_of_ne_zero`. I didn't see a simple term-mode proof like you suggest, did I miss something?

Reply:
I think `this_lemma.trans $ or_iff_left h` will work?

Reply:
Nice one, thanks!